src/main/java/seedu/address/logic/parser/FilterCommandParser.java

```diff
@@ -37,37 +44,182 @@ public FilterCommand parse(String args) throws ParseException {
                 ArgumentTokenizer.tokenize(args, PREFIX_TIME, PREFIX_INGREDIENT_GRAIN, PREFIX_INGREDIENT_VEGE,
                         PREFIX_INGREDIENT_PROTEIN, PREFIX_INGREDIENT_OTHER, PREFIX_GOAL);
 
-        List<Time> filterByTime;
+        boolean filterByFavourites = filterByFavourites(argMultimap);
+        if (!argMultimap.getPreamble().equals("")) {
```

Comment:
Im not sure whether it is what you are going for, but could  (!argMultimap.getPreamble().isBlank()) be better?

Reply:
I honestly didn't know such a method existed. Thank you! Will update it.

As for what I am going for, I am checking for the presence of invalid keywords (ie. any word _other than_ "favourites").

I will shift this check into the the `filterByFavourites(argMultimap)` as it is clearer that way.